### PR TITLE
chore(deps): Upgrade to Webpack 4

### PR DESCRIPTION
**Pull request by jstoffan**
Todo
- [x] Manual testing of regression folder

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

File: build/webpack.config.js [hunk 1]
```diff
@@ -24,90 +23,74 @@ if (fs.existsSync('build/rsync.json')) {
...
         entry: {
             annotations: ['box-annotations'],
             preview: [`${lib}/Preview.js`],
-            csv: [`${lib}/viewers/text/BoxCSV.js`]
+            csv: [`${lib}/viewers/text/BoxCSV.js`],
+        },
+        mode: isRelease ? 'production' : 'development',
+        optimization: {
+            minimizer: [
+                new UglifyJsPlugin({
```
Comment (ConradJChan) [inline build/webpack.config.js]: I had read that TerserPlugin (which I think comes by default) is preferred. Should we switch to that instead?

Comment (jstoffan) [reply build/webpack.config.js]: I've seen that as well, but it seemed beyond the scope of what I was trying to achieve. It seems to work fine and other repos all still use UglifyJsPlugin, so I'm going to leave it as-is for now.

Comment (ConradJChan) [reply build/webpack.config.js]: 👍

**Pull request merged**